syntax = "proto3";

package broadcast;

message HpMpUpdate {
  int32 currentHp = 1;
  int32 maxHp = 2;
  int32 currentMp = 3;
  int32 maxMp = 4;
}

message CharacterLevelUpdate {
  int32 level = 1;
}

message CharacterExperienceUpdate {
  uint64 experience = 1;
}

message CharacterNameUpdate {
  string name = 1;
}

message BroadcastMessage {
  oneof body {
    HpMpUpdate hpMpUpdate = 1;
    CharacterLevelUpdate characterLevelUpdate = 2;
    CharacterExperienceUpdate characterExperienceUpdate = 3;
    CharacterNameUpdate characterNameUpdate = 4;
  }
}